src/Model/Profile.php

```diff
@@ -307,7 +307,7 @@ public static function getVCardHtml(array $profile, bool $block, bool $show_cont
 
 		$profile_url = $profile['url'];
 
-		$cid = $profile['id'];
+		$cid = Contact::getIdForURL($profile['url']);
```

Comment:
You could use `getPublicAndUserContactID` instead.

Reply:
Please just tell me which way is the least costly.

Reply:
You could use 
```
$contact = Contact::selectFirst(['uid*], ['uri-id` => $profile['uri-id], 'uid' => 0]};
```

Reply:
Thanks!